### PR TITLE
Add inlay hints for showing omitted function return types

### LINKED ISSUE #869: Inlay hints: function type/effects
Related to #524 
Extracted from #828
Blocked on #833

Here's some code that fits into `override def getInlayHints` in `effekt/jvm/src/main/scala/effekt/Server.scala` and that mostly worked locally:
```scala
   // TODO: Also do this for un-annotated `val`s and `var`s.
    val unannotated = getTreesInRange(range)(using context).map { trees =>
      trees.collect {
        // Functions without an annotated type:
        case fun: FunDef if fun.ret.isEmpty => for {
          // TODO(question): `context.inferredTypeAndEffectOption(fun)` didn't work here. Why?
          sym <- context.symbolOption(fun.id)
          tpe <- context.functionTypeOption(sym)
          pos <- positions.getFinish(fun.bparams) // TODO(bug): This position doesn't always work. Ideally, I'd use the position of `fun.ret`, but that doesn't exist!
        res = InlayHint(InlayHintKind.Type, pos, pp": ${tpe.result} / ${tpe.effects}", resolveToLabel = true, paddingLeft = true, effektKind = "missing-type-annotation")
        } yield res
      }.flatten
    }.getOrElse(Vector())

    val allHints = captures ++ unannotated
    if allHints.isEmpty then None else Some(allHints)
```

### PR DESCRIPTION
**Pull request by timsueberkrueb**
Fixes #869. The attached text edits also suffer from #1001.

### CODE REVIEW

**Comment by timsueberkrueb** [comment]
If you like, I can also add them for the other def kinds in this PR.

File: effekt/jvm/src/main/scala/effekt/Server.scala [hunk 1]
```diff
@@ -397,6 +397,32 @@ class Server(config: EffektConfig, compileOnChange: Boolean=false) extends Langu
...
+            } yield {
+              val prettyType = pp": ${tpe.result} / ${tpe.effects}"
+              val inlayHint = new InlayHint(convertPosition(pos), messages.Either.forLeft(pp": ${tpe.result} / ${tpe.effects}"))
+              inlayHint.setKind(InlayHintKind.Type)
+              val markup = new MarkupContent()
+              markup.setValue(s"return type${prettyType}")
+              markup.setKind("markdown")
+              inlayHint.setTooltip(markup)
+              inlayHint.setPaddingLeft(true)
+              inlayHint.setData("omitted-type-annotation")
```
Comment (jiribenes) [inline effekt/jvm/src/main/scala/effekt/Server.scala:417]: Philosophically, I think it would be wise to name this as to what it adds, not what it resolves. :)
(I'm willing to compromise about this, but let's argue first if you feel strongly about `omitted-`)
```suggestion
              inlayHint.setData("return-type-annotation")
```
(don't forget the tests too)

File: effekt/jvm/src/main/scala/effekt/Server.scala [hunk 2]
```diff
@@ -397,6 +397,32 @@ class Server(config: EffektConfig, compileOnChange: Boolean=false) extends Langu
...
+        val unannotated = getTreesInRange(range)(using context).map { trees =>
+          trees.collect {
+            // Functions without an annotated type:
+            case fun: FunDef if fun.ret.isEmpty => for {
+              sym <- context.symbolOption(fun.id)
+              tpe <- context.functionTypeOption(sym)
+              pos = fun.ret.span.range.from
+            } yield {
+              val prettyType = pp": ${tpe.result} / ${tpe.effects}"
+              val inlayHint = new InlayHint(convertPosition(pos), messages.Either.forLeft(pp": ${tpe.result} / ${tpe.effects}"))
```
Comment (jiribenes) [inline effekt/jvm/src/main/scala/effekt/Server.scala:410]: ?
```suggestion
              val prettyType = pp": ${tpe.result} / ${tpe.effects}"
              val inlayHint = new InlayHint(convertPosition(pos), messages.Either.forLeft(prettyType))
```

**Review by jiribenes — COMMENTED** [review]
Mostly LGTM, could you also add one tests which shows an _effect_ being returned in the annotation (besides the empty effect set)?

**Review by jiribenes — APPROVED** [review]
LGTM :shipit:
